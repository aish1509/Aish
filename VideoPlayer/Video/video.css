video::-webkit-media-controls {
  display: none !important;
}
video {
  width:100%;
  height:100%;
  min-height:250px;
}