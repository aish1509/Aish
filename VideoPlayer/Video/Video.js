import React from 'react';
import PropTypes from 'prop-types';
import './video.css';

export default function Video({
	children,
	video,
	onPlay,
	onPause,
	onEnded,
	onSeeking,
	onSeeked,
	onCanPlay,
	handleMetaData,
	handleTimeUpdate,
	handleContextMenu,
	poster
}) {
	return (
		<video
			ref={video}
			controlsList="nodownload"
			id="v"
			width="100%"
			height="auto"
			crossOrigin="anonymous"
			onPlay={onPlay}
			onPause={onPause}
			onEnded={onEnded}
			onSeeking={onSeeking}
			onSeeked={onSeeked}
			onCanPlay={onCanPlay}
			onLoadedMetadata={handleMetaData}
			onTimeUpdate={handleTimeUpdate}
			onContextMenu={handleContextMenu}
			poster={poster}
		>
			{children}
		</video>
	);
}

Video.defaultProps = {
	children: React.node,
	onEnded: () => null,
	handleContextMenu: () => null
};
Video.propTypes = {
	children: PropTypes.node,
	video: PropTypes.object.isRequired,
	onPlay: PropTypes.func.isRequired,
	onPause: PropTypes.func.isRequired,
	onEnded: PropTypes.func,
	onSeeking: PropTypes.func.isRequired,
	onSeeked: PropTypes.func.isRequired,
	onCanPlay: PropTypes.func.isRequired,
	handleMetaData: PropTypes.func.isRequired,
	handleTimeUpdate: PropTypes.func.isRequired,
	handleContextMenu: PropTypes.func,
	poster: PropTypes.string
};
