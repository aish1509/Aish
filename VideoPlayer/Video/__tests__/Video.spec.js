import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Video from '../index';

describe('<Video />', () => {
	let wrapper = null;
	let handleTimeUpdateMock = null;
	let handleMetaDataMock = null;
	let onCanPlayMock = null;
	let onSeekedMock = null;
	let onPlayMock = null;
	let onPauseMock = null;
	let onEndedMock = null;
	let onSeekingMock = null;
	let video = null;

	beforeEach(() => {
		handleTimeUpdateMock = jest.fn();
		handleMetaDataMock = jest.fn();
		onCanPlayMock = jest.fn();
		onSeekedMock = jest.fn();
		onSeekingMock = jest.fn();
		onPlayMock = jest.fn();
		onPauseMock = jest.fn();
		handleTimeUpdateMock = jest.fn();
		handleTimeUpdateMock = jest.fn();
		video = React.createRef();

		wrapper = shallow(
			<Video
				video={video}
				handleTimeUpdate={handleTimeUpdateMock}
				handleMetaData={handleMetaDataMock}
				onCanPlay={onCanPlayMock}
				onSeeked={onSeekedMock}
				onPlay={onPlayMock}
				onPause={onPauseMock}
				onEnded={onEndedMock}
				onSeeking={onSeekingMock}
			/>
		);
	});

	it('should render correctly', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('video callbacks fires corretly', () => {
		const videoRef = React.createRef();
		wrapper = mount(
			<Video
				video={videoRef}
				handleTimeUpdate={handleTimeUpdateMock}
				handleMetaData={handleMetaDataMock}
				onCanPlay={onCanPlayMock}
				onSeeked={onSeekedMock}
				onPlay={onPlayMock}
				onPause={onPauseMock}
				onEnded={onEndedMock}
				onSeeking={onSeekingMock}
			/>
		);

		const VideoComp = wrapper.find('video');
		expect(VideoComp.length).toBe(1);

		const result = Video.defaultProps.onEnded();
		expect(result).toBe(null);

		const result2 = Video.defaultProps.handleContextMenu();
		expect(result2).toBe(null);

		VideoComp.getElement().ref.current.dispatchEvent(new Event('play'));
		expect(onPlayMock).toHaveBeenCalled();
		expect(onPlayMock.mock.calls.length).toEqual(1);

		VideoComp.getElement().ref.current.dispatchEvent(new Event('pause'));
		expect(onPauseMock).toHaveBeenCalled();
		expect(onPauseMock.mock.calls.length).toEqual(1);

		VideoComp.getElement().ref.current.dispatchEvent(new Event('seeking'));
		expect(onSeekingMock).toHaveBeenCalled();
		expect(onSeekingMock.mock.calls.length).toEqual(1);

		VideoComp.getElement().ref.current.dispatchEvent(new Event('seeked'));
		expect(onSeekedMock).toHaveBeenCalled();
		expect(onSeekedMock.mock.calls.length).toEqual(1);

		VideoComp.getElement().ref.current.dispatchEvent(
			new Event('loadedmetadata')
		);
		expect(handleMetaDataMock).toHaveBeenCalled();
		expect(handleMetaDataMock.mock.calls.length).toEqual(1);

		VideoComp.getElement().ref.current.dispatchEvent(new Event('canplay'));
		expect(onCanPlayMock).toHaveBeenCalled();
		expect(onCanPlayMock.mock.calls.length).toEqual(1);

		VideoComp.getElement().ref.current.dispatchEvent(
			new Event('timeupdate')
		);
		expect(handleTimeUpdateMock).toHaveBeenCalled();
		expect(handleTimeUpdateMock.mock.calls.length).toEqual(1);

		VideoComp.getElement().ref.current.dispatchEvent(
			new Event('contextmenu')
		);

		wrapper.unmount();
	});
});
