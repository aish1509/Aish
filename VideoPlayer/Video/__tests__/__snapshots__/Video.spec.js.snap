// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> should render correctly 1`] = `
<video
  controlsList="nodownload"
  crossOrigin="anonymous"
  height="auto"
  id="v"
  onCanPlay={[MockFunction]}
  onContextMenu={[Function]}
  onEnded={null}
  onLoadedMetadata={[MockFunction]}
  onPause={[MockFunction]}
  onPlay={[MockFunction]}
  onSeeked={[MockFunction]}
  onSeeking={[MockFunction]}
  onTimeUpdate={[MockFunction]}
  width="100%"
/>
`;
